src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -82,6 +82,35 @@ public IEnumerable<CompletionItem> GetFilteredCompletions(Compilation compilatio
                 .Concat(GetDisableNextLineDiagnosticsDirectiveCodesCompletion(model, context));
         }
 
+
+        public IEnumerable<CompletionItem> GetFilteredParamsCompletetions(ParamsSemanticModel paramsSemanticModel, ParamsCompletionContext paramsCompletionContext)
+        {
+            //1 - bicep semantic model -> parameter declarations
+            //2 - once param keyword is typed -> suggest parameter completions
+                
+        
+            //param  
+            return Enumerable.Empty<CompletionItem>().Concat(GetParamAssingmnetCompletions(paramsSemanticModel, paramsCompletionContext));
+        }
+
+        private IEnumerable<CompletionItem> GetParamAssingmnetCompletions(ParamsSemanticModel paramsSemanticModel, ParamsCompletionContext paramsCompletionContext)
+        {   
+            if(paramsCompletionContext.Kind.HasFlag(ParamsCompletionContextKind.ParamAssignment) && paramsSemanticModel.bicepCompilation is {} bicepCompilation)
+            {
+                var bicepSemanticModel = bicepCompilation.GetEntrypointSemanticModel();
+                var bicepFileParamDeclarations = bicepSemanticModel.Root.ParameterDeclarations;
+
+                //TODO: handle the case where there are no params in bicep file but we are trying to write one in params file
+                if(bicepFileParamDeclarations.Length != 0)
+                {
```

Comment:
You can skip this check; if the list is empty, the body of the `foreach` loop will never be entered.